tests/Agent/IntegrationTests/Shared/OracleConfiguration.cs

```diff
@@ -82,20 +72,43 @@ public static string OraclePort
         {
             get
             {
-                if (_oraclePort == null)
+                EnsureHostPortParsed();
+                return _oraclePort;
+            }
+        }
+
+        // Ensures host and port are parsed and cached only once
+        private static void EnsureHostPortParsed()
+        {
+            if (!_parsedHostPort)
+            {
+                try
                 {
-                    try
-                    {
-                        var uri = new UriBuilder(OracleDataSource);
-                        _oraclePort = uri.Port.ToString();
-                    }
-                    catch (Exception ex)
-                    {
-                        throw new Exception("OraclePort configuration is invalid.", ex);
-                    }
+                    ParseHostAndPort(OracleDataSource, out _oracleServer, out _oraclePort);
+                    _parsedHostPort = true;
+                }
+                catch (Exception ex)
+                {
+                    throw new Exception("OracleServer or OraclePort configuration is invalid.", ex);
                 }
+            }
+        }
 
-                return _oraclePort;
+        // Parses host and port from a data source string like host:port/service
+        private static void ParseHostAndPort(string dataSource, out string host, out string port)
+        {
+            // Regex matches host (hostname or IPv4), port, and ignores service name
+            var match = Regex.Match(dataSource, @"^(?<host>[^:/\[]+(?:\.[^:/\[]+)*)[:](?<port>\d+)");
```

Comment:
I'm interested in hearing what other team members think, but I feel like using a regex here is more complicated than we need it to be. We could probably get away with something similar to what was done previously by doing something like `new UriBuilder("oracle://" + dataSource)`.

Reply:
That's probably true, but it's only test code, and I think it's clear enough from the context what it's doing.

Reply:
Refactored to eliminate regex